src/equations/svaerd_kalisch_1d.jl

```diff
@@ -176,21 +176,22 @@ function create_cache(mesh, equations::SvaerdKalischEquations1D,
     beta_hat = equations.beta * D .^ 3
     gamma_hat = equations.gamma * sqrt.(equations.gravity * D) .* D .^ 3
     tmp2 = similar(h)
+    M = mass_matrix(solver.D1)
     if solver.D1 isa PeriodicDerivativeOperator ||
        solver.D1 isa UniformPeriodicCoupledOperator
         D1_central = solver.D1
         sparse_D1 = sparse(D1_central)
-        D1betaD1 = sparse_D1 * Diagonal(beta_hat) * sparse_D1
+        minus_MD1betaD1 = sparse_D1' * M * Diagonal(beta_hat) * sparse_D1
     elseif solver.D1 isa PeriodicUpwindOperators
         D1_central = solver.D1.central
-        D1betaD1 = sparse(solver.D1.plus) * Diagonal(beta_hat) * sparse(solver.D1.minus)
+        sparse_D1minus = sparse(solver.D1.minus)
+        minus_MD1betaD1 = sparse_D1minus' * M * Diagonal(beta_hat) * sparse_D1minus
```

Comment:
```suggestion
        # We use the periodic upwind SBP property
        #   M * sparse_D1plus == -sparse_D1minus' * M
        # to avoid possible floating point errors in the symmetry of the matrix.
        sparse_D1minus = sparse(solver.D1.minus)
        minus_MD1betaD1 = sparse_D1minus' * M * Diagonal(beta_hat) * sparse_D1minus
```